V1 and V2-alpha deprecation help docs (#54)
beaker/docs: docs/concept/breaking-changes.md

  arguments: [ arguments that you need to specify to go along with commands ]
  envvars: 
    name: env var name
    value: optional source the environment variable from a literal value
    secret: optional source the environment variable from a secret in the experiment's workspace.
  datasets:
    mountPath:  absolute path within the container to mount the data source
    subPath: optional sub path to a file or directory within a mounted data source
    source:
      # pick exactly one of:
      beaker: beaker data set by name or ID
      hostPath: source data from your host path
      result: source data from a previous task by name
      url: source data from a cloud service provider
      secret: source data from a sercret in the same workspace as the experiment
  result:
    path: path to where the task will write result output
  resources:
    cpuCount: ex - 4, .5
    gpuCount: ex - 1, 5
    memory: ex - 2.5 GiB
    sharedMemory: ex - 2.5 GiB (default is 5 GiB)
  context:
    cluster: required name or ID of a cluster on which the task should run
    priority: urgent, high, normal (normal is the default), low 

> ```

Reviewer: What do you think of showing each change in isolation, with a diff (github markdown supports this format). For instance, something like:

### Task Result Path

The result path is no longer a child of the `spec` field, and is now a dictionary:

```diff
  - spec:
       resultPath: /results
  - result:
      path: /results
```

This might be a bit of a pain to produce, but I think I think it'd be the best experience for our users. We could divide and conquer here of that's helpful so you don't have to write all of this yourself!

You may notice that the authorToken, dependsOn, and minimum memory requirement fields have been deprecated.

> @sam or michal, what should I say about why we're deprecating authorToken and dependsOn??

Author: @codeviking or @aimichal

Reviewer: I think `dependsOn` was replaced by the ability to link tasks via dataset dependencies. There's an example of how to do so [here](https://github.com/beaker/docs/tree/main/examples/pipeline). So mentioning that it has been deprecated (maybe in a `Deprecated Properties` section?) and linking to that example seems like it'd be good.

I'm not sure what `authorToken` was. By looking at the code it looks like it's a way of aliasing result datasets and groups (though I'm not sure why groups are relevant) to some another Beaker user (via their Beaker token).  Maybe it was a way of running something as Beaker on behalf of someone else.  Anyway I highly doubt it's in use, so I'd just mention that the field is no longer supported.

> Minimum memory requirement fields are being deprecated because memory will not be preacllocated. You may set a maximum limit for the amount of memory your experiments will use.

Reviewer: ```suggestion
Minimum memory requirement fields are being deprecated because memory is no longer preallocated. You may set a maximum limit for the amount of memory your experiments will use.
```

> 

Author: i think i should also add a "what's new" section. Thoughts?

Reviewer: Sure. 

I wonder if we should make this a "Release Notes" of sort, and name the file as such.

So maybe the doc would be `docs/release-notes.md` and we'd add a section for each major release, or something.

```
# Release Notes

The release notes here are incomplete, and only capture major changes. 

## v1.4.44

### What's New

...

### Breaking Changes

...

## v1.3.38
```

Reviewer: Also, I imagine (and hope) that we'll somehow make this more of an automated thing that's a part of something like the new beaker-docs effort in the future!

## Questions?

> Feel free to reach out to us on the [Beaker Users Slack Channel](https://allenai.slack.com/archives/C6MN19S05)

Author: is this a slack channel i can put on public facing docs..?

Reviewer: I think that's fine. But I tend to think most things are fine. @dplessas Do you have any concerns about a link to a Slack channel in public documentation?

Reviewer 2: No that is fine.
